resolwe/flow/models/entity.py

```diff
@@ -58,27 +59,59 @@ class Meta(BaseCollection.Meta):
             ("owner_relation", "Is owner of the relation"),
         )
 
+    UNIT_HOUR = 'hr'
+    UNIT_CHOICES = (
+        (UNIT_HOUR, 'Hour'),
+    )
+
+    def validate_unique(self, exclude=None):
+        """Check unique constraints on the model.
+
+        Separate check for the combination of ``collection`` and ``category``
+        is required as ``category`` is optional and thus
```

Comment:
Could all this be avoided by just providing a default value for `category`and then use `Meta.unique_together`?

Reply:
Yeah, I really don't like this. Remind me why do we have to support NULL category? Could we have an empty string by default?

Reply:
`NULL` means that you are not using categories, so you can have multiple relations of the same type. Setting a default value (e.g. empty string) would restrain this, which is, in my opinion, not a good option.

Alternative would be to make the category mandatory.

Reply:
> Alternative would be to make the category mandatory.

Let's do that. I would not like the to offer no-category alternative to the users.